### PR TITLE
Add Passportjs

### LINKED ISSUE #1333: [Add] Passport.js
### Site URL

https://www.passportjs.org/

### Description of the above provided site

**Passport.js**

Passport.js is an authentication middleware that is used with nodejs application to add various authentication ways like sign in with Google or sign in in with GitHub etc on websites.

### Category

Backend

### Sub Category

Authentication

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by AlfiyaSiddique**
## Fixes Issue
Issue #1333 
closes #1333

## Changes proposed
Add the Passport authentication middleware in the backend authentication category

## Note to reviewers
Please merge this pr with gssoc23 labels

### CODE REVIEW

File: database/backend/authentication.json [hunk 1]
```diff
@@ -19,5 +19,12 @@
     "url": "https://appwrite.io/docs/authentication",
     "category": "backend",
     "subcategory": "authentication"
+  },
+  {
+    "name": "Passport Authentication",
+    "description": "Passport is authentication middleware for Node.js. Extremely flexible and modular, Passport can be unobtrusively dropped in to any Express-based web application. A comprehensive set of strategies support authentication using a username and password, Facebook, Twitter, and more.",
```
Comment (CBID2) [inline database/backend/authentication.json:25]:
```suggestion
    "description": "Passport is authentication middleware for Node.js. Extremely flexible and modular, this tool can be unobtrusively dropped in to any Express-based web application. It also provides a comprehensive set of strategies support authentication for a username and password, Facebook, Twitter, and more.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @AlfiyaSiddique. I provided a suggestion on how to improve your description.

**Comment by AlfiyaSiddique** [comment]
@CBID2 Thanks for the suggestions. 👍🏻
Improved the description

File: database/backend/authentication.json [hunk 2]
```diff
@@ -19,5 +19,12 @@
     "url": "https://appwrite.io/docs/authentication",
     "category": "backend",
     "subcategory": "authentication"
+  },
+  {
+    "name": "Passport Authentication",
+    "description": "Passport is authentication middleware for Node.js. Extremely flexible and modular, this tool can be unobtrusively dropped in to any Express-based web application. It also provides a comprehensive set of strategies support authentication for a username and password, Facebook, Twitter, and more.",
```
Comment (CBID2) [inline database/backend/authentication.json:25]: [Proofreading]
```suggestion
    "description": "Passport is authentication middleware for Node.js. Extremely flexible and modular, this tool can be unobtrusively dropped in to any Express-based web application. It also provides a comprehensive set of support authentication strategies for usernames, passwords, Facebook, Twitter, and more.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Made another suggestion for improvement

**Comment by AlfiyaSiddique** [comment]
Did the changes @CBID2

**Review by CBID2 — APPROVED** [review]
LGTM! :)

**Comment by AlfiyaSiddique** [comment]
> LGTM! :)

Yeah, then can you please merge this PR!!

**Comment by CBID2** [comment]
> > LGTM! :)
> 
> 
> 
> Yeah, then can you please merge this PR!!

Two approvals are required for PRs to be merged. Please be patient.

**Review by k-deepak04 — APPROVED** [review]

**Pull request merged**